webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -82,25 +83,31 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
                 .filename(file.getOriginalFilename());
 
         final MessageDigest messageDigest = DigestUtils.getMd5Digest();
-        final List<CsvValidationFailure> validationFailures = new ArrayList<>();
+        final List<CsvValidationResult> validationResults = new ArrayList<>();
 
         try {
             try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
-                validationFailures.addAll(validate(digestStream, permissionScope));
+                validationResults.addAll(validate(digestStream, permissionScope));
             }
 
             final String digest = Hex.encodeHexString(messageDigest.digest()).toUpperCase();
-            if (validationFailures.isEmpty()) {
+            List<CsvValidationResult> failures = validationResults
+                    .stream()
+                    .filter(x -> !x.isOk())
+                    .collect(Collectors.toList());
```

Comment:
I suppose technically I could just look at the first one as they should be the same, but just for thorough defensiveness in case something slips through in a later code update?